pytorch_lightning/utilities/argparse.py

```diff
@@ -286,6 +286,13 @@ def _gpus_allowed_type(x) -> Union[int, str]:
         return int(x)
 
 
+def _gpus_arg_default(x) -> Union[int, str]:  # pragma: no-cover
+    # unused, but here for backward compatibility with old checkpoints that need to be able to
+    # unpickle the function from the checkpoint, as it was not filtered out in versions < 1.2.8
+    # see: https://github.com/PyTorchLightning/pytorch-lightning/pull/6898
+    return _gpus_allowed_type(x)
```

Comment:
Should we just?:

```suggestion
    pass
```

Reply:
means return None?

Reply:
yes

Reply:
I'm trying to figure out if that would break something. sigh

In the other PR #6898 , I removed this code
```python
# nothing was passed into the GPUs argument
    if callable(gpus):
        return None
```

So I think when loading the checkpoint, we should directly replace this legacy function with None.

Reply:
@awaelchli tests are green and you have the approvals. Do you want to do this or keep it as is?

Reply:
Ok, I will test once again that I can load the old checkpoint, but with this change.